gettsim/typing.py

```diff
@@ -17,17 +18,35 @@ def check_if_series_has_internal_type(series, internal_type):
 
     Returns
     -------
-    out : bool
-        Return check variable.
+    out : adjusted pd.Series and boolean
     """
     if internal_type == float:
         out = is_float_dtype(series) or is_integer_dtype(series)
+        if out:
+            return (out, series)
+        else:
+            series = series.astype(float, errors="ignore")
+            return (is_float_dtype(series) or is_integer_dtype(series), series)
     elif internal_type == bool:
         out = is_bool_dtype(series)
+        if out:
+            return (out, series)
+        else:
+            series = series.astype(bool, errors="ignore")
+            return (is_bool_dtype(series), series)
```

Comment:
- Better practise to have only one return statement at the very end of the function
- No need to return `is_bool_dtype(series)`. The function should always return the series with the expected type so no need to return or check the type.